hal/armv7m/cpu.h

```diff
@@ -16,6 +16,16 @@
 #ifndef _HAL_CPU_H_
 #define _HAL_CPU_H_
 
+
+#if defined(CPU_STM32L152XD) || defined(CPU_STM32L152XE) || defined(CPU_STM32L4X6)
+#define CPU_STM32
```

Comment:
We could do #include "stm32.h" here

Reply:
I don't understand why we need this CPU_STM32 macro. I suggest to define TARGET macro in CFLAGS definition. This macro should be based on the same notation as TARGET definition i.e. TARGET_ARMV7M4_STM32L152XD for armv7m4-stm32l52xd.

Reply:
`We could do #include "stm32.h" here`
I tried to include **stm32.h** in this header. However it has occurred that **stm32.h** includes **cpu.h** and there is a loop with headers. It is the main reason why I divided it.

`I don't understand why we need this CPU_STM32 macro`
CPU_STM32 macro is used in many places in kernel i.e.: programs.c, proc/threads.c, syscalls.c. We can get rid of it but the code of the kernel has to be changed. I didn't want to make such big changes in a single PR.

`I suggest to define TARGET macro in CFLAGS definition.`
Ok.